### PR TITLE
0.15 Release Notes: `Set` Reflection

### PR DESCRIPTION
**Pull request by RobWalt**
Write the section mentioned in #1665

### CODE REVIEW

File: release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md
Comment (cart) [inline release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md:38]: EntityHashSet is not generic. Perhaps it would be better to make this `HashSet<Entity>`, rather than using our built in / special cased `EntityHashSet`, just for simplicity?

File: release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md
Comment (MrGVSV) [inline release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md:17]: I'm not sure how others feel about this but I feel like we should be highlighting the new features provided by `Set` rather than explain how it will affect a single crate (not that we _can't_ mention 3rd party crates, just that I don't think it should be the main focus)

File: release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md
Comment (MrGVSV) [inline release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md:20]: Nit: This example might be better served showcasing how `Set` works, maybe even comparing it to `Map`.

Or perhaps if we do want to use bevy-inspector-egui here, maybe we could show a screenshot of this in action?

**Comment by alice-i-cecile** [comment]
@RobWalt are you planning to get back to this in the next couple of days?

File: release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md
Comment (RobWalt) [reply release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md:17]: I'm actually not really sure about further implications since I'm not super familiar with bevy reflect. My main motivation for this change was to unblock bevy-inspector-egui after I had a deep dive into the issue.

File: release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md
Comment (RobWalt) [reply release-content/0.15/release-notes/13014_Dedicated_Reflect_implementation_for_Setlike_things.md:20]: Re: Screenshots: I had the same thought and wanted to do before/after screenshots but `Set` also has to be implemented from the bevy-inspector-egui side first. I wanted to create a pre-release PR there, but there are a lot of bigger structural changes and not just adjustments because of the new `Set` variant, so I kind of stopped because I don't feel competent enough.

**Comment by RobWalt** [comment]
@alice-i-cecile We might as well skip this section in the release notes if it blocks anything, no hard feelings. I actually didn't think that the PR was big enough for something like notes in the first place. It only was supposed to unblock bevy_inspector_egui and I didn't really think about other use cases yet.

**Comment by alice-i-cecile** [comment]
@SkiFire13 @MrGVSV I've added you both as contributors due to your incredibly valuable help during review. @RobWalt I've shortened this substantially to keep this simple and clear. The people who care about this will be very excited but most users will go "hmm that's kinda neat I guess". If you have a moment, take a look and give me your LGTM?

**Review by MrGVSV — APPROVED** [review]

**Review by cart — APPROVED** [review]

**Pull request merged**
